godot-macros/src/lib.rs

```diff
@@ -444,6 +446,35 @@ use crate::util::ident;
 /// #     fn init(base: godot::obj::Base<Self::Base>) -> Self { todo!() }
 /// # }
 /// ```
+///
+/// <div class="stab portability">Available on <strong>crate feature <code>register-docs</code></strong> only.</div>
+/// <div class="stab portability">Available on <strong>Godot version <code>4.3+</code></strong> only.</div>
+///
+/// # Documentation
+///
+/// You can document your functions, classes, members, and signals with the `///` doc comment syntax.
+///
+/// ```no_run
+/// # use godot::prelude::*;
+/// #[derive(GodotClass)]
+/// # #[class(init)]
+/// /// This is an example struct for documentation, inside documentation.
+/// struct DocumentedStruct {
+///     /// This is a class member.
+///     /// You can use markdown formatting such as _italics_.
+///     #[var]
+///     item: f32,
+/// }
+///
+/// #[godot_api]
+/// impl DocumentedStruct {
+///     /// This provides the item, after adding `0.2`.
+///     #[func]
+///     pub fn produce_item(&self) -> f32 {
+///         self.item + 0.2
+///     }
+/// }
+/// ```
```

Comment:
The `#[class(init)]` can be visible, otherwise a user copy-pasting this will run into compile errors.

Reply:
Why does the property registration section hide the class(init)?

Reply:
I don't know, maybe we should fix this separately. Or you can also keep `# #[class(init)]` hidden for now and we address this later, it's a small issue.